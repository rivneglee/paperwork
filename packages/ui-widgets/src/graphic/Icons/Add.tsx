/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
      <path d="M965.846914 563.832099H58.153086C29.708642 563.832099 6.320988 540.444444 6.320988 512s23.387654-51.832099 51.832098-51.832099h907.693828c28.444444 0 51.832099 23.387654 51.832098 51.832099s-23.387654 51.832099-51.832098 51.832099z"></path>
      <path d="M512 1017.679012c-28.444444 0-51.832099-23.387654-51.832099-51.832098V58.153086C460.167901 29.708642 483.555556 6.320988 512 6.320988s51.832099 23.387654 51.832099 51.832098v907.693828c0 28.444444-23.387654 51.832099-51.832099 51.832098z"></path>
  </svg>
);
