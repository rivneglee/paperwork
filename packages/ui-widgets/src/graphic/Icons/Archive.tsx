/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1024 1024" version="1.1">
      <path
        d="M438.857143 219.428571V146.285714H365.714286v73.142857h73.142857z m73.142857 73.142858V219.428571H438.857143v73.142858h73.142857zM438.857143 365.714286V292.571429H365.714286v73.142857h73.142857z m73.142857 73.142857V365.714286H438.857143v73.142857h73.142857z m400-221.714286q16 16 27.428571 43.428572t11.428572 50.285714v658.285714q0 22.857143-16 38.857143t-38.857143 16H128q-22.857143 0-38.857143-16t-16-38.857143V54.857143q0-22.857143 16-38.857143t38.857143-16h512q22.857143 0 50.285714 11.428571t43.428572 27.428572z m-253.714286-139.428571v214.857143h214.857143q-5.714286-16.571429-12.571428-23.428572l-178.857143-178.857143q-6.857143-6.857143-23.428572-12.571428z m219.428572 873.142857V365.714286h-237.714286q-22.857143 0-38.857143-16t-16-38.857143V73.142857H512v73.142857H438.857143V73.142857H146.285714v877.714286h731.428572zM519.428571 538.857143l61.142858 199.428571q4.571429 15.428571 4.571428 29.714286 0 47.428571-41.428571 78.571429T438.857143 877.714286t-104.857143-31.142857T292.571429 768q0-14.285714 4.571428-29.714286 12-36 68.571429-226.285714V438.857143h73.142857v73.142857h45.142857q12.571429 0 22.285714 7.428571t13.142857 19.428572z m-80.571428 265.714286q30.285714 0 51.714286-10.857143t21.428571-25.714286-21.428571-25.714286-51.714286-10.857143-51.714286 10.857143-21.428571 25.714286 21.428571 25.714286 51.714286 10.857143z"
      ></path>
  </svg>
);
