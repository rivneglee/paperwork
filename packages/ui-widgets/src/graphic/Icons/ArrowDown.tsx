/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
    <path
      d="M846.496 379.733333a52 52 0 0 0-71.552 0L506.965333 637.866667 238.933333 379.733333a52 52 0 0 0-71.466666 0 47.413333 47.413333 0 0 0 0 68.928L471.189333 741.333333a52 52 0 0 0 71.552 0l303.754667-292.629333a47.509333 47.509333 0 0 0 0-68.970667z"
      ></path>
    <path
      d="M506.965333 758.229333a53.877333 53.877333 0 0 1-37.472-14.933333L165.728 450.698667a49.76 49.76 0 0 1-15.68-36.458667 50.293333 50.293333 0 0 1 15.68-36.458667c20.661333-19.904 54.282667-19.904 74.954667 0l266.282666 256.501334 266.282667-256.501334c20.661333-19.904 54.293333-19.904 74.954667 0a50.304 50.304 0 0 1 0 72.917334L544.437333 743.296a53.877333 53.877333 0 0 1-37.472 14.933333zM203.2 368.192a49.002667 49.002667 0 0 0-34.08 13.578667 44.8 44.8 0 0 0-14.016 32.469333 44.298667 44.298667 0 0 0 14.005333 32.448L472.885333 739.306667a49.568 49.568 0 0 0 68.16 0L844.8 446.709333a44.693333 44.693333 0 0 0 0-64.938666 49.557333 49.557333 0 0 0-68.16 0L506.965333 641.546667 237.28 381.770667a49.002667 49.002667 0 0 0-34.08-13.578667z"
      ></path>
    <path
      d="M506.965333 758.816a54.666667 54.666667 0 0 1-38.165333-15.221333L165.034667 450.997333a50.186667 50.186667 0 0 1-15.84-36.757333 50.72 50.72 0 0 1 15.84-36.757333c21.045333-20.266667 55.285333-20.266667 76.330666 0l265.6 255.84 265.6-255.84c21.045333-20.266667 55.285333-20.266667 76.330667 0a50.709333 50.709333 0 0 1 0 73.525333L545.12 743.605333a54.656 54.656 0 0 1-38.154667 15.210667zM203.2 368.778667a48 48 0 0 0-33.386667 13.301333 44.384 44.384 0 0 0-13.866666 32.16 43.872 43.872 0 0 0 13.866666 32.149333L473.6 739.008a48.565333 48.565333 0 0 0 66.784 0L844.117333 446.4a44.362667 44.362667 0 0 0 0-64.330667 48.565333 48.565333 0 0 0-66.784 0L506.965333 642.517333 236.586667 382.069333a48 48 0 0 0-33.386667-13.290666z"
      ></path>
    <path
      d="M506.965333 761.514667A57.098667 57.098667 0 0 1 467.093333 745.6L163.338667 452.992a52.896 52.896 0 0 1-16.672-38.752A53.461333 53.461333 0 0 1 163.338667 375.466667c21.984-21.173333 57.749333-21.173333 79.722666 0l263.904 254.218666L770.858667 375.466667c21.984-21.173333 57.749333-21.173333 79.722666 0a53.504 53.504 0 0 1 0 77.514666L546.826667 745.6a57.098667 57.098667 0 0 1-39.861334 15.914667zM203.2 364.949333a52.426667 52.426667 0 0 0-36.458667 14.528 48 48 0 0 0-15.008 34.762667 47.434667 47.434667 0 0 0 14.986667 34.752l303.776 292.618667a53.045333 53.045333 0 0 0 72.928 0l303.765333-292.608a47.893333 47.893333 0 0 0 0-69.536c-20.106667-19.370667-52.821333-19.370667-72.928 0L506.965333 636.949333l-267.306666-257.472a52.426667 52.426667 0 0 0-36.458667-14.528z m303.765333 390.026667a50.453333 50.453333 0 0 1-35.093333-13.973333l-303.786667-292.629334a46.592 46.592 0 0 1-14.677333-34.133333 47.125333 47.125333 0 0 1 14.698667-34.133333 50.976 50.976 0 0 1 70.176 0L506.965333 638.933333l268.672-258.858666a51.029333 51.029333 0 0 1 70.176 0 47.168 47.168 0 0 1 0 68.266666L542.048 741.002667a50.453333 50.453333 0 0 1-35.082667 13.973333zM203.2 371.466667a45.386667 45.386667 0 0 0-31.690667 12.608 41.6 41.6 0 0 0-13.024 30.165333 41.141333 41.141333 0 0 0 12.992 30.133333l303.797334 292.64a46.090667 46.090667 0 0 0 63.381333 0l303.754667-292.608a41.6 41.6 0 0 0 0-60.341333 46.090667 46.090667 0 0 0-63.381334 0l-272.064 262.08-272.074666-262.069333a45.386667 45.386667 0 0 0-31.690667-12.608z"
      ></path>
  </svg>
);
