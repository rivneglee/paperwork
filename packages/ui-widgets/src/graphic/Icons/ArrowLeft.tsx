/* tslint:disable:max-line-length */
import * as React from 'react';

export default ({ className = '' }: { className?: string }) => (
  <svg className={`pw-icon ${className}`} viewBox="0 0 1070 1024">
    <path d="M788.224 168.362667L339.456 617.130667 233.813333 511.658667 682.666667 62.805333z"
          p-id="5748"></path>
    <path d="M683.008 960.341333L234.24 511.658667l105.642667-105.728 448.682666 448.853333z"
          p-id="5749"></path>
  </svg>
);
