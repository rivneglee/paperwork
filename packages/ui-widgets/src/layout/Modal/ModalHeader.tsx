import React, { FunctionComponent, ReactElement } from 'react';

interface Props {
  icon?: ReactElement;
  title: string | ReactElement;
  subTitle?: string | ReactElement;
}

export type ModalHeaderComponent = FunctionComponent<Props>;

const ModalHeader: ModalHeaderComponent = ({
  title,
  subTitle,
  icon,
}) => (
  <div className="pw-modal-dialog__header">
    <div className="pw-modal-dialog__title">
      {
        icon && <span className="pw-modal-dialog__header-icon">{icon}</span>
      }
      {title}
    </div>
    {
      subTitle && <div className="pw-modal-dialog__sub-title">{subTitle}</div>
    }
  </div>
);

export default ModalHeader;
