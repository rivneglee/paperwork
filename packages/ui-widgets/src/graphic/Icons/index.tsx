import Add from './Add';
import Chart from './Chart';
import Document from './Document';
import Form from './Form';
import Menu from './Menu';
import ArrowLeft from './ArrowLeft';
import ArrowRight from './ArrowRight';
import ArrowUp from './ArrowUp';
import ArrowDown from './ArrowDown';
import QuestionMark from './QuestionMark';
import Info from './Info';
import Unlink from './Unlink';
import BlockQuote from './BlockQuote';
import Bold from './Bold';
import H2 from './H2';
import OL from './OL';
import Italic from './Italic';
import Close from './Close';
import Link from './Link';
import UL from './UL';
import DataSource from './DataSource';
import Template from './Template';
import Logo from './Logo';
import Search from './Search';
import Trash from './Trash';
import Save from './Save';
import Delete from './Delete';
import Cancel from './Cancel';
import Calendar from './Calendar';
import Duplicate from './Duplicate';
import Confirm from './Confirm';
import Error from './Error';
import PowerOff from './PowerOff';
import Edit from './Edit';
import Settings from './Settings';
import DragHandle from './DragHandle';
import Theme from './Theme';
import Layout from './Layout';
import Control from './Control';
import Text from './Text';
import TextInput from './TextInput';
import ComboBox from './ComboBox';
import Attachment from './Attachment';
import Page from './Page';
import Rate from './Rate';
import TextArea from './TextArea';
import List from './List';
import None from './None';
import Columns from './Columns';
import Rows from './Rows';
import PieChart from './PieChart';
import LineChart from './LineChart';
import Preview from './Preview';
import Button from './Button';
import Submit from './Submit';
import Dashboard from './Dashboard';
import Target from './Target';
import Timer from './Timer';
import WeChat from './WeChat';
import Facebook from './Facebook';
import Twitter from './Twitter';
import LinkedIn from './LinkedIn';
import WeBlog from './WeBlog';
import Share from './Share';
import Archive from './Archive';
import Commit from './Commit';
import QRCode from './QRCode';
import Pass from './Pass';

export default {
  Add,
  ArrowLeft,
  ArrowRight,
  ArrowUp,
  ArrowDown,
  Chart,
  Document,
  Form,
  Menu,
  QuestionMark,
  Info,
  Unlink,
  BlockQuote,
  H2,
  OL,
  Italic,
  Close,
  Bold,
  Link,
  UL,
  DataSource,
  Template,
  Logo,
  Search,
  Trash,
  Save,
  Duplicate,
  Delete,
  Cancel,
  Confirm,
  Error,
  PowerOff,
  Edit,
  Settings,
  DragHandle,
  Theme,
  Layout,
  Control,
  Text,
  TextInput,
  ComboBox,
  Attachment,
  Page,
  Rate,
  TextArea,
  List,
  None,
  Columns,
  Rows,
  PieChart,
  LineChart,
  Preview,
  Button,
  Submit,
  Dashboard,
  Target,
  Timer,
  Calendar,
  WeChat,
  LinkedIn,
  Facebook,
  Twitter,
  WeBlog,
  Share,
  Archive,
  Commit,
  QRCode,
  Pass,
};
