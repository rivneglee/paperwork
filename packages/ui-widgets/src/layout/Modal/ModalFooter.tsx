import React, { FunctionComponent } from 'react';
import { Separator } from '../../graphic/Separator';

interface Props {}

export type ModalFooterComponent = FunctionComponent<Props>;

const ModalFooter: ModalFooterComponent = ({
  children,
}) => (
  <div className="pw-modal-dialog__footer">
    <Separator />
    {children}
  </div>
);

export default ModalFooter;
