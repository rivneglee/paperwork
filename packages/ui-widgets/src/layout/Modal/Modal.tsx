import React, { FunctionComponent, ReactElement } from 'react';
import ReactModal from 'react-modal';
import classNames from 'classnames';
import ModalHeader, { ModalHeaderComponent } from './ModalHeader';
import ModalFooter, { ModalFooterComponent } from './ModalFooter';

interface Props {
  isOpen: boolean;
  className?: string;
  onClose?: () => void;
  footer: ReactElement;
  header: ReactElement;
}

interface ModalComponent extends FunctionComponent<Props>{
  Header: ModalHeaderComponent;
  Footer: ModalFooterComponent;
}

const Modal: ModalComponent = ({
  isOpen,
  className,
  children,
  onClose,
  footer,
  header,
}) => (
  <ReactModal
    isOpen={isOpen}
    onRequestClose={onClose}
    ariaHideApp={false}
    className={classNames('pw-modal-dialog', className)}
    overlayClassName="pw-modal-dialog__overlay">
    {header}
    {
      children && <div className="pw-modal-dialog__content">{children}</div>
    }
    {footer}
  </ReactModal>
);

Modal.Header = ModalHeader;
Modal.Footer = ModalFooter;

export default Modal;
