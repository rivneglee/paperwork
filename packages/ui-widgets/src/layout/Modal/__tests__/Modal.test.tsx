import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';

import Modal from '../Modal';
import Icons from '../../../graphic/Icons';

describe('Modal', () => {
  const Content = () => <>Content</>;

  it('should render modal when it not opened', () => {
    const wrapper: ShallowWrapper = shallow(
      <Modal
        header={<Modal.Header title="foo"/>}
        footer={<Modal.Footer>Footer</Modal.Footer>}
        isOpen={false}>
        <Content />
      </Modal>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render modal when it opened', () => {
    const wrapper: ShallowWrapper = shallow(
      <Modal
        isOpen={true}
        header={<Modal.Header title="foo"/>}
        footer={<Modal.Footer>Footer</Modal.Footer>}
      >
        <Content />
      </Modal>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render modal with header icon', () => {
    const wrapper: ShallowWrapper = shallow(
      <Modal
        isOpen={true}
        header={<Modal.Header title="foo" icon={<Icons.Confirm/>}/>}
        footer={<Modal.Footer>Footer</Modal.Footer>}
      >
        <Content />
      </Modal>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
