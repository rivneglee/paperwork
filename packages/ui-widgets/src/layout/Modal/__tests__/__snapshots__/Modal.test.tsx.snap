// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render modal when it not opened 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="pw-modal-dialog"
  closeTimeoutMS={0}
  isOpen={false}
  overlayClassName="pw-modal-dialog__overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalHeader
    title="foo"
  />
  <div
    className="pw-modal-dialog__content"
  >
    <Content />
  </div>
  <ModalFooter>
    Footer
  </ModalFooter>
</Modal>
`;

exports[`Modal should render modal when it opened 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="pw-modal-dialog"
  closeTimeoutMS={0}
  isOpen={true}
  overlayClassName="pw-modal-dialog__overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalHeader
    title="foo"
  />
  <div
    className="pw-modal-dialog__content"
  >
    <Content />
  </div>
  <ModalFooter>
    Footer
  </ModalFooter>
</Modal>
`;

exports[`Modal should render modal with header icon 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="pw-modal-dialog"
  closeTimeoutMS={0}
  isOpen={true}
  overlayClassName="pw-modal-dialog__overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <ModalHeader
    icon={<Unknown />}
    title="foo"
  />
  <div
    className="pw-modal-dialog__content"
  >
    <Content />
  </div>
  <ModalFooter>
    Footer
  </ModalFooter>
</Modal>
`;
